### DISCUSSION TITLE
Does self heal work while argocd is performing a sync

### DISCUSSION THREAD

**Discussion by SarthakNarayan** [argoproj/argo-cd#17439] [Q&A]
So I have a microservices application and I am using argocd for GitOps. My application has a specific upgrade procedure that uses sync waves and requires one of the pods to be shut down. But I am afraid if I try to scale the replica count to 0 argocd will bring it back to the desired count and break the update procedure.

Are there any work arounds for this? I don't want to turn self heal off and on just for the upgrade process.

**Comment by switchboardOp** [comment]
I tested this today and to me it looked like auto-heal changed the replicas back immediately. 

I set up a preSync hook that ran `kubectl scale deployment --replicas=0` on another workload defined in the application. 

It could have been that they sync was just very fast though.
